### PR TITLE
Minify uglify code

### PR DESCRIPTION
**Pull request by matheusrocha89**
Huge changes to make the plugin better:
- [x] Making tests run on travis
- [x] Splitting directives, factories and controllers each one in one file
- [x] Refactoring code to use a little bit of angular style guide
- [x] Creating gulp tasks to concat and minify
- [x] Reverting the pr #22 because it was breaking the player

My next PR I will configure bower of the project, then when the user installs with bower it will only download the files on dist folder not the whole project.

### CODE REVIEW

**Comment by arielfaur** [comment]
Hi @matheusrocha89, nice work!  I only find the code hard to follow with this structure. The minifying and uglifying tasks are great but I am having trouble following the code to make changes...seems I am not used to this syntax. Seems too verbose to me.  Any chances we can use the older syntax? Is it that bad?

**Comment by matheusrocha89** [comment]
The only problem I saw on the older syntax it's the idea of having one big file with everything. This approach let open chances to put all the code in just one big file and later it will be hard to maintain. The idea here is to isolate each directive and service in your specific file and your specific responsibility, like angular styleguide, but it's just a PR :smile: if you don't like this approach no problem :smile:

**Comment by arielfaur** [comment]
The idea of splitting files is fine, I see the benefit. Just the code style
using the $inject instead of Inline Array Annotation. In fact Angular docs
suggest inline is the preferred way. See here:
https://docs.angularjs.org/guide/di/#dependency-annotation. By using inline
and closures -instead of separate functions- the code seems easier to read.
What do you think?

**Comment by matheusrocha89** [comment]
For me no problem, if you want this style it's ok.

**Comment by matheusrocha89** [comment]
Just change the injection to inline?

**Comment by arielfaur** [comment]
I can do that myself :)

**Comment by matheusrocha89** [comment]
:+1: :smile: Thanks
